### PR TITLE
Remove using environment variables in CMakeLists.

### PR DESCRIPTION
**Pull request by ye-luo**
Following #365 
QMCPACK specific ones are removed.
CC, CXX and library specific ones are managed by CMake.

CXX, C_FLAGS, CXX_FLAGS are no more used explicitly by our CMakeLists.

Since CC, CXX is still supported by CMake. There is no need to update the README and the manual.

### CODE REVIEW

**Comment by ye-luo** [comment]
@prckent In the ctest_script.cmake, ENV{N_PROCS} is still kept.

**Comment by ye-luo** [comment]
OK to test

File: CMakeLists.txt [hunk 1]
```diff
@@ -361,13 +338,11 @@ ELSE(CMAKE_TOOLCHAIN_FILE)
   #------------------------------------
   # Identify the compiler
   #------------------------------------
-  IF( $ENV{CXX} MATCHES "icpc" )
-    SET( COMPILER Intel )
-  ELSEIF($ENV{CXX} MATCHES "xlC" OR CMAKE_C_COMPILER_ID MATCHES "XL")
-    SET( COMPILER IBM )
-  ELSEIF ( CMAKE_C_COMPILER_WORKS OR CMAKE_C_COMPILER_WORKS )
+  IF ( CMAKE_C_COMPILER_WORKS OR CMAKE_C_COMPILER_WORKS )
     IF( CMAKE_COMPILE_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX )
```
Comment (prckent) [inline CMakeLists.txt]: While we are updating: I suggest updating this section to 

1.  Use CMAKE_C_COMPILER_ID and CMAKE_CXX_COMPILER_ID consistently, as per the Intel check, for GNU and IBM. See https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html
2. Remove the microsoft check. We know of no one using this in years and it is a totally unchecked configure path. We have an "unknown compiler" route.

**Comment by prckent** [comment]
OK to test

**Comment by ye-luo** [comment]
Include a change for #368

**Pull request merged**